src/main/java/com/ericsson/ei/rules/MatchIdRulesHandler.java

```diff
@@ -67,7 +73,10 @@ public String replaceIdInRules(String matchIdString, String id) {
         if (matchIdString.contains(replacementMarker)) {
             return matchIdString.replace(replacementMarker, id);
         } else {
-            return null;
+            String errorMessage = String.format("MatchIdRules: %s does not contain the rules.replacement.marker: %s", matchIdString, replacementMarker);
+            ReplacementMarkerException exception = new ReplacementMarkerException(errorMessage);
+            LOGGER.warn("", exception);
```

Comment:
Will this exception not be logget where it is catched?

Reply:
No, for some reason spring does not want to log this runtime exception. The solution was to implement your own HandlerExceptionResolver which sounded like overkill.

Reply:
Maybe put this information in a comment so other readers of the code knows what is going on.

Reply:
This is handled differently now.